application/models/artifact_builder.py

```diff
@@ -19,26 +20,16 @@ def __init__(self):
 
     def build_with(self, **properties):
         """ builds self.neo and saves it"""
-        neo_properties = self._make_properties_neo_compatible(**properties)
+        props = PropertyBuilder(**properties)
+        neo_properties = props.node_properties
+        print(neo_properties)
         self.neo = Artifact(**neo_properties).save()
-        if 'tags' in properties:
-            self._add_tags_to_neo(properties['tags'])
-        if 'team_id' in properties and properties.get('team_id') is not None:
-            self._connect_to_team(properties['team_id'])
-        properties['id'] = self.neo.id_
+        print(self.neo)
+        self._set_full_tags(props.tag_properties, override_tags=False)
+        if 'team_id' in props.properties and props.properties.get('team_id') is not None:
+            self._connect_to_team(props.properties['team_id'])
```

Comment:
A better structuring of all this would be a classification into `node_properties` and `relationship_properties` And then have a ArtifactNodeBuilder (or just a method for that since it is mostly constructor calling) and have an ArtifactRelationshipBuilder  that works with the relationship_properties and adds the tags, team and in the future author via different methods.

Thoughts on that design?

Reply:
Sounds good, look at the new property_handler whether you like the changes.